### DISCUSSION TITLE
Can a autogenerated sidebar be used in versioned documentations?

### DISCUSSION THREAD

**Discussion by PhilippHeuer** [facebook/docusaurus#7924] [Q&A] [answered]
Hi.

The docs on `https://docusaurus.io/docs/versioning` show an example which supports `.json` for sidebar configurations in versioned documentations.

In my case this would be:

```md
├── versioned_sidebars
│   ├── version-v2-sidebars.json
│   └── version-v1-sidebars.json
```

Is there a way to use `autogenerate` sidebars in versioned docs? This seems to be possible in .js files, but i couldn't find anything for the .json sidebar config file. (or a way to use .js files instead of the .json files)

**Comment by PhilippHeuer** [comment, marked as answer]
Figured out that the json allows multiple different formats, in addition to the simple one i saw in all examples there is another one with a more complex format that allows to set the type to autogenerated:

```json
{
  "version-v1-docs": [
    {
      "type": "autogenerated",
      "dirName": "."
    }
  ]
}

```

**Reply by slorber** [reply]
> Figured out that the json allows multiple different formats

No really it's the exact same format. The only difference it that versioned sidebars are .json and the current docs sidebar is .js. Otherwise the data structure allowed is exactly the same. When versioning your sidebar, we just execute the js file and save it as json, nothing more.
